ebean-core/src/main/java/io/ebeaninternal/server/query/SqlTreeBuilder.java

```diff
@@ -311,7 +311,7 @@ private SqlTreeNode buildNode(String prefix, STreePropertyAssoc prop, STreeType
 
     } else {
       // do not read Id on child beans (e.g. when used with fetch())
-      boolean withId = isNotSingleAttribute();
+      boolean withId = isNotSingleAttribute() && !subQuery;
```

Comment:
Note: in Ebean 11 I had two cases to fix: https://github.com/FOCONIS/ebean/commit/403d1e80987f2609b40d02e88a4f1344079780e9#diff-5cf825762bcb8ca9a200d86e0882dcd1d740715fcaf39872cf4e98595edd3177R336

both cases covered by a test case. Tests are passed in Ebean 12 without the second modification

Reply:
Comes from this change: https://github.com/FOCONIS/ebean/pull/26. PR & Test will follow when this is merged